doc/designs/edns.md

```diff
@@ -0,0 +1,198 @@
+
+This initial version of DoT is supported only for new installations/deployments, although administrators can run `ipa-dns-install` with the new DoT options to redeploy the integrated DNS service with DoT enabled on an existing deployment.
+
+The deployment of the DNS service will depend on the infrastructure configuration. During server/replica/client deployment or client/replica enrollment, both unencrypted and encrypted communications will be allowed as default. However, for more restrictive scenarios where only encrypted communications are allowed, a new option is needed to enforce this behavior. The new option `--dns-policy` will have two choices: `relaxed` will allow unencrypted DNS queries, while `enforced` will restrict unencrypted queries on port 53.
+
+The implementation of DoT in FreeIPA's DNS service leverages the following design choices:
+
+- Current FreeIPA works with `Bind` 9.18. This version of BIND doesn't support DoT-forwarding. Therefore, for server/replica deployment, we rely on an additional component, `unbound`, which redirects encrypted DNS queries to the external DNS server. This includes enabling TLS connections on TCP port 853 and managing TLS certificates for both components `Bind` and `Unbound`. The upgrade to the latest LTS `Bind` version (9.20) is already in progress but not completed. Once this upgrade is complete, the design will be simplified by removing `unbound` as Bind will fully support DoT-forwarding. 
+
+- Configuration Management: New LDAP attributes and FreeIPA CLI options are introduced to enable and configure DoT settings.
+
+- Certificate Integration: FreeIPA's certificate management framework can automatically be used to generate and manage TLS certificates giving the administrators the choice to either provide a certificate or rely on custodia for certificate auto enrollment if no certificate is provided. These options are handled by the proper CLI options (please check section [CLI](#cli)).
+
+- Service Discovery: Clients/Replicas can detect DoT-capable DNS servers directly during the installation process. However, it is important to mention that prior to use `enforced` policy as part of replica or client enrollment, the administrator must pre-configure the systems via NetworkManager. The goal is to ensure that the client installation process can securely discover IPA servers when `--dns-policy=enforced` is used. This limitation will be addressed in the future by implementing additional automation by modifying `ipa-client-install` to configure Unbound as a DoT forwarder before performing domain discovery and ensure that the forwarder is used to resolve SRV/TXT records for IPA server discovery.
```

Comment:
I'd simply this from "prior to use" to "before using enforce policy...". And clarify what in NetworkManager that needs to be pre-configured. Is this "limitation" mentioned that manual config via NetworkManager is required? That isn't clear. Also you mentioned in the previous paragraph that unbound would not be required once bind 1.20 is ready . Does that also apply here?